cmake/projects/Boost/schemes/url_sha1_boost_library.cmake.in

```diff
@@ -188,8 +188,16 @@ else()
   set(copy_mpi_command "")
 endif()
 
+if(BUILD_SHARED_LIBS) #if using shared libs overide boost config
+  set(Boost_USE_STATIC_LIBS OFF)
```

Comment:
The variable name is a little bit confising, `set(Boost_USE_STATIC_LIBS OFF)` will be visible only in this scope (i.e. for building external project) but not in the user's scope (the file that generated from BoostConfig.cmake.in).

I think `BoostConfig.cmake.in` should looks like that:

```cmake
if(@BUILD_SHARED_LIBS@) # BUILD_SHARED_LIBS from external project build step
  set(Boost_USE_STATIC_LIBS OFF) # variable visible for user
else()
  set(Boost_USE_STATIC_LIBS ON)
endif()
```